src/StackYaml/Transformations/UpdateExtraDeps.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+module StackYaml.Transformations.UpdateExtraDeps
+    ( transformation
+    ) where
+
+import Control.Applicative         (many)
+import Control.Exception           (IOException, catch)
+import Control.Lens
+import Data.Aeson.Lens
+import Data.Char                   (isDigit)
+import Data.Function               ((&))
+import Data.List                   (intersperse)
+import Data.Monoid                 ((<>))
+import Data.Text                   (Text, pack)
+import Data.Yaml                   (Value)
+import System.Exit                 (exitFailure)
+import System.IO                   (hPutStrLn, stderr)
+import Text.Regex.Applicative.Text (anySym, few, match, psym, sym)
+
+import           Distribution.Package  (PackageIdentifier (..),
+                                        PackageName (..))
+import qualified Distribution.PackDeps as PackDeps
+import           Distribution.Version  (Version (..))
+
+transformation :: Value -> IO Value
+transformation value = do
+    newest <- catch PackDeps.loadNewest
+                    (\e -> hPutStrLn stderr (failedToLoad e) *> exitFailure)
+    traverseOf
+      (key "extra-deps" . values . _String)
+      (updateExtraDep newest)
+      value
+  where
+    failedToLoad :: IOException -> String
+    failedToLoad err =
+      unlines
+        [ show err
+        , "Failed to read package database. Doing 'cabal update' might help."
+        ]
+
+    updateExtraDep :: PackDeps.Newest -> Text -> IO Text
+    updateExtraDep newest dep =
+        (pkgName <$> parsePackageIdentifier dep)
+           & maybe
+               (dep <$ hPutStrLn
+                         stderr
+                         ("Ignoring invalid extra-deps entry " <> show dep))
+               (\name ->
+                   newestPackageId newest name
+                     & maybe
+                         (dep <$ hPutStrLn
+                                   stderr
+                                   ("Ignoring extra-deps entry "
+                                       <> show dep
+                                       <> " as it refers to unknown package "
+                                       <> show (unPackageName name)))
+                         (pure . pPrintPackageIdentifier))
```

Comment:
I probably should just use `case` here, to be consistent with [`updateGithubDep`](https://github.com/phadej/stack-yaml/pull/2/files#diff-9cddc95048f32deafa05519a80386aa5R21)

Or we could rewrite `updateGithubDep` as:

```haskell
updateGithubDep :: Object -> IO Object
updateGithubDep obj =
    ((obj ^? ix "git" . _String) >>= parseGithub)
      & maybe
          (pure obj)
          (\(owner, repo) ->
              (GH.executeRequest' $ GH.branchesForR owner repo $ Just 1)
                 & either
                     (\err -> throwM err)
                     (\branches ->
                        (V.find ((== "master") . GH.branchName) branches)
                           & maybe
                               (pure obj)
                               (\branch ->
                                   let obj' = obj & ix "commit" . _String .~ GH.branchCommitSha)
                                   in pure obj'))
```